integrationtests/src/test/java/io/kroxylicious/proxy/filter/OutOfBandSendFilter.java

```diff
@@ -0,0 +1,80 @@
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.common.protocol.Errors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.config.BaseConfig;
+
+import static io.kroxylicious.UnknownTaggedFields.unknownTaggedFieldsToStrings;
+
+/**
+ * Designed to work in tandem with another Filter like {@link RequestResponseMarkingFilter}. This Filter targets
+ * DescribeClusterRequest and sends an out-of-band request instead. Then, when it gets a response via the sendRequest
+ * CompletionStage, it collects the Unknown Tagged Fields for a specified tag and forwards them as an error message in
+ * a DescribeClusterResponse. This exposes what unknown tagged fields were added to the out-of-band response by other
+ * filters like {@link RequestResponseMarkingFilter}.
+ */
+public class OutOfBandSendFilter implements DescribeClusterRequestFilter {
+
+    private final OutOfBandSendFilterConfig config;
+
+    public OutOfBandSendFilter(OutOfBandSendFilterConfig config) {
+        this.config = config;
+    }
+
+    public static class OutOfBandSendFilterConfig extends BaseConfig {
+        private final ApiKeys apiKeyToSend;
+        private final int tagIdToCollect;
+
+        @JsonCreator
+        public OutOfBandSendFilterConfig(@JsonProperty(value = "apiKeyToSend", required = true) ApiKeys apiKeyToSend,
+                                         @JsonProperty(value = "tagToCollect", required = true) int tagIdToCollect) {
+            this.apiKeyToSend = apiKeyToSend;
+            this.tagIdToCollect = tagIdToCollect;
+        }
+
+    }
+
+    @Override
+    public void onDescribeClusterRequest(short apiVersion, RequestHeaderData header, DescribeClusterRequestData request, KrpcFilterContext context) {
+        ApiKeys apiKeyToSend = config.apiKeyToSend;
+        ApiMessage message = createApiMessage(apiKeyToSend);
+        context.sendRequest(apiKeyToSend.latestVersion(), message).thenAccept(apiMessage -> {
+            String values = unknownTaggedFieldsToStrings(apiMessage, config.tagIdToCollect).collect(Collectors.joining(","));
+            // WARNING: forwarding from within the completion is not generally supported yet, this only works because the test is strictly controlling
+            // requests being sent to Kroxylicious. If used in a proxy between a real Kafka client and broker, this could cause out-of-order responses to
+            // the client, causing client side errors. We use it here so that we can see the outcome of the sendRequest call with a single request to Kroxylicious.
+            context.forwardResponse(
+                    new DescribeClusterResponseData().setErrorCode(Errors.UNKNOWN_SERVER_ERROR.code())
+                            .setErrorMessage("filterNameTaggedFieldsFromOutOfBandResponse: " + values));
```

Comment:
What would it take to expose this in a compliant fashion or to make this response model work?

Reply:
To expose it in a compliant fashion is a little work since the mock server can only handle one mock response currently. If the mock server could support multiple responses like: `respond to CREATE_TOPIC with X, respond to DESCRIBE_CLUSTER with Y`, then we could:
1. intercept describe cluster, send and out-of-band request and forward on the original describe cluster
2. out-of-band request would be sent and received first, so from completionstage we can store the result in a field on the filter
3. describe cluster response received from the server and we could intercept it and set the error message

To make the response model work we'd need a solution to https://github.com/kroxylicious/kroxylicious/issues/351, which amounts to queueing up requests/responses somewhere to ensure we send them to the upstream broker in order and respond to the client in order, regardless of what order the filters call `forwardRequest|forwardResponse`.

Reply:
> If the mock server could support multiple responses like: respond to CREATE_TOPIC with X, respond to DESCRIBE_CLUSTER with Y

Certainly sounds like a reasonable enhancement. Maybe we should track that and leave note to re-factor this usage once that is done?

Reply:
https://github.com/kroxylicious/kroxylicious/issues/460